scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala

```diff
@@ -12,43 +12,77 @@ class RedundantParens(implicit ctx: RewriteCtx) extends RewriteSession {
 
   import ctx.dialect
 
-  def isWrappedInParens(tokens: Tokens): Boolean =
-    tokens.nonEmpty && tokens.head.is[LeftParen] && tokens.last.is[RightParen]
-
   override def rewrite(tree: Tree): Unit =
     tree match {
-      case g: Enumerator.Guard =>
-        val tokens: Tokens = g.cond.tokens
-        if (isWrappedInParens(tokens)) {
-          val openParensCount = tokens.segmentLength(_.is[LeftParen], 0)
-          val closeParensCount =
-            tokens.reverse.segmentLength(_.is[RightParen], 0)
-          val parensToRemoveCount = Math.min(openParensCount, closeParensCount)
-          val leftSegment: Tokens = tokens.slice(0, parensToRemoveCount)
-          val rightSegment: Tokens =
-            tokens.slice(tokens.length - parensToRemoveCount, tokens.length)
-          leftSegment.zip(rightSegment.reverse).foreach {
-            case (left, right) if ctx.isMatching(left, right) =>
-              ctx.addPatchSet(TokenPatch.Remove(left), TokenPatch.Remove(right))
-            case _ =>
-          }
-        }
+      case t: Defn.Val => maybeRemove(t.rhs)
 
-      case t @ Term.Apply(_, List(b: Term.Block))
-          if ctx.style.activeForEdition_2020_01 &&
-            b.tokens.headOption.exists(_.is[Token.LeftBrace]) =>
-        t.tokens.lastOption.filter(_.is[RightParen]).foreach { rparen =>
-          ctx.getMatchingOpt(rparen).foreach { lparen =>
-            implicit val builder = Seq.newBuilder[TokenPatch]
-            builder += TokenPatch.Remove(lparen)
-            builder += TokenPatch.Remove(rparen)
-            ctx.removeLFToAvoidEmptyLine(lparen)
-            ctx.removeLFToAvoidEmptyLine(rparen)
-            ctx.addPatchSet(builder.result(): _*)
-          }
-        }
+      case t: Defn.Def => maybeRemove(t.body)
+
+      case t => rewriteFunc.applyOrElse(t, (x: Tree) => remove(x, true))
+    }
+
+  private val rewriteFunc: PartialFunction[Tree, Unit] = {
+    case g: Enumerator.Guard => remove(g.cond)
+
+    case t: Case if t.cond.isDefined => remove(t.cond.get)
```

Comment:
```suggestion
    case t: Case => t.cond.foreach(remove)
```

Reply:
the reason i did it so is to make sure this partial function wouldn't match (elsewhere, if the partial didn't match, i would do something else).

in this specific, we can "pretend" it matched as there's probably nothing else that can be done in case of... well, `case`. let me know your preference.

Reply:
i tested, no change in behaviour, yours is better (don't like `.get` on `Option`).